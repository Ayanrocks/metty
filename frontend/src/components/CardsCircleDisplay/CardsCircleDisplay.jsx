import React from 'react';

export default function CardsCircleDisplay(props) {
  const { weatherData } = props;
  return (
    <div
      style={{
        margin: '32px auto',
        zIndex: 2,
      }}
    >
      <div
        style={{
          padding: '20px',
          width: '350px',
          height: '350px',
          textAlign: 'center',
          boxShadow: '0 2px 5px rgba(0,0,0,0.4)',
          backgroundColor: '#fff',
          opacity: '57%',
          zIndex: 2,
          display: 'flex',
          flexDirection: 'column',
        }}
      >
        <div>
          <h1
            style={{
              fontSize: 32,
              fontWeight: 500,
              color: '#000',
            }}
          >
            {weatherData.name}
          </h1>
        </div>
        <div
          style={{
            height: '120px',
            margin: '-20px 0 0 0',
          }}
        >
          <p
            style={{
              fontSize: 84,
              color: '#000',
            }}
          >
            {weatherData.temp}&#176;
            <span style={{ fontSize: 64 }}>C</span>
          </p>
        </div>
        <div
          style={{
            display: 'flex',
            flexWrap: 'wrap',
            justifyContent: 'space-evenly',
          }}
        >
          <div>
            <p style={{ fontSize: 16, color: '#000' }}>
              Feels Like: {weatherData.feelsLike}&#176;C
            </p>
          </div>
          <div>
            <p style={{ fontSize: 16, color: '#000' }}>
              Humidity: {weatherData.humidity}%
            </p>
          </div>

          <div>
            <p style={{ fontSize: 16, color: '#000' }}>
              Cloudiness: {weatherData.cloudiness}%
            </p>
          </div>
        </div>
        <div>
          <p style={{ fontSize: 25, color: '#000' }}>
            {weatherData.weatherType}
          </p>
        </div>
      </div>
    </div>
  );
}
