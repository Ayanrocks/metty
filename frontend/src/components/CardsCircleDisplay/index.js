import CardsCircleDisplay from './CardsCircleDisplay';

export default CardsCircleDisplay;
