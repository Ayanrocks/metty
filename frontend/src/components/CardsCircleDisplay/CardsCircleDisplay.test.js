import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CardsCircleDisplay from '.';

Enzyme.configure({ adapter: new Adapter() });

describe('CardsCircleDisplay component', () => {
  test('it renders and matches with snapshot', () => {
    const weatherMock = {
      name: 'Kolkata',
      temp: 24,
      humidity: 60,
      feelsLike: 28,
      cloudiness: 40,
      weatherType: 'Haze',
      weatherIcon: `http://openweathermap.org/img/w/u50n.png`,
    };
    const wrapper = shallow(<CardsCircleDisplay weatherData={weatherMock} />);
    expect(wrapper).toMatchSnapshot();
  });
});
