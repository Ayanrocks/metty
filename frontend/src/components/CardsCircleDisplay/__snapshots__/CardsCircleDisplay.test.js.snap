// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsCircleDisplay component it renders and matches with snapshot 1`] = `ShallowWrapper {}`;
